packages/babel-helper-create-class-features-plugin/src/fields.ts

```diff
@@ -273,6 +274,13 @@ const privateNameHandlerSpec: Handler<PrivateNameState & Receiver> & Receiver =
             ? "classStaticPrivateMethodGet"
             : "classStaticPrivateFieldSpecGet";
 
+        const binding = member.scope.getBinding(classRef.name);
+        if (innerBinding && binding && !(binding.identifier === innerBinding)) {
+          // the classRef has been shadowed
+          throw binding.path.buildCodeFrameError(
+            `Shadowing class ${classRef.name} with private property`,
+          );
```

Comment:
> Q: any idea how could we generate correct code?
> Should we rename the local variable?

Yeah, I'd rename the local variable:
```js
path.scope.rename(classRef.name);
```

Reply:
Sure!

I renamed all the local variables that shadowing the `classRef` including those in parent scopes.